src/local-history/local-history-manager.ts

```diff
@@ -266,6 +280,8 @@ export class LocalHistoryManager {
                 vscode.window.showInformationMessage(`'${path.basename(revision.fsPath)}' was deleted.`);
                 vscode.commands.executeCommand(Commands.TREE_REFRESH);
             });
+
+            this.removeEmptyFoldersRec();
```

Comment:
I think this should be moved into the `unlink` (perhaps before we show the info message.